services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceControllerClusterConfig.java

```diff
@@ -996,6 +1007,13 @@ public VeniceControllerClusterConfig(VeniceProperties props) {
     this.pushJobUserErrorCheckpoints = parsePushJobUserErrorCheckpoints(props);
     this.isHybridStorePartitionCountUpdateEnabled =
         props.getBoolean(ConfigKeys.CONTROLLER_ENABLE_HYBRID_STORE_PARTITION_COUNT_UPDATE, false);
+
+    this.helixRebalancePreferenceEvenness = props.getInt(CONTROLLER_HELIX_REBALANCE_PREFERENCE_EVENNESS, -1);
+    this.helixRebalancePreferenceLessMovement = props.getInt(CONTROLLER_HELIX_REBALANCE_PREFERENCE_LESS_MOVEMENT, -1);
+    this.helixRebalancePreferenceForceBaselineConverge =
+        props.getInt(CONTROLLER_HELIX_REBALANCE_PREFERENCE_FORCE_BASELINE_CONVERGE, -1);
+    this.helixInstanceCapacity = props.getInt(CONTROLLER_HELIX_INSTANCE_CAPACITY, -1);
+    this.helixResourceCapacityWeight = props.getInt(CONTROLLER_HELIX_RESOURCE_CAPACITY_WEIGHT, -1);
```

Comment:
Can we have the basic validations here instead of in `ZkHelixAdminClient` so that we can fail fast on controller startup rather than identifying issues only midway through new cluster creation. Also, instead of exposing these raw config values, we can expose materialized objects instead. Like: `helixGlobalRebalancePreference`, `helixInstanceCapacityKeys`, `helixDefaultInstanceCapacityMap`, `helixDefaultPartitionWeightMap`, etc.

Reply:
And we don't have to expose the `-1` sentinel values that are unsupported by Helix out of this config interface